src/main.rs

```diff
@@ -95,25 +92,28 @@ async fn main() {
     info!("Found content topics for subscription: {:?}", topics);
 
     debug!("Initializing Graphcast Agent");
+
+    let graphcast_agent_config = GraphcastAgentConfig {
```

Comment:
maybe there could be a radio Config function that converts the radio specific config to GraphcastAgentConfig? It's not necessary but would be cleaner?

Reply:
the Radio config could grow a lot and be more complex, I would really prefer to decouple them

Reply:
my point is based on the thought that not all fields in Radio config should be `GraphcastAgentConfigs`, but Radio config must include all fields in `GraphcastAgentConfig`, such that whatever Radio config is defined, they should have a function that, convert might be the wrong word, but derive GraphcastAgentConfig from the Radio Config